src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -850,11 +850,13 @@ string BuildCommentsString(const SourceLocation& location,
   string final_comments;
   string epilogue;
 
-  if (prefer_single_line && lines.size() == 1) {
+  bool is_one_line = prefer_single_line && lines.size() == 1;
+
+  if (is_one_line) {
     prefix = "/** ";
     suffix = " */\n";
   } else {
-    prefix = " * ";
+    prefix = "* ";
```

Comment:
ok, instead of overloading is_one_line, add bool add_leading_space and set it to true here and false in the oneline case, and test that where you do the prefix.  Then things are a little more clear.